android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -66,6 +65,68 @@ public Screen getScreen() {
     return mScreenView;
   }
 
+  public void onContainerUpdate() {
+    mHasChildWithConfig = hasChildScreenWithConfig(getScreen());
+    markParentFragments(mHasChildWithConfig);
+    if (!mHasChildWithConfig) {
+      // if there is no child with config, we look for a parent with config to set the orientation
+      ScreenStackHeaderConfig config = findHeaderConfig();
+      if (config != null && config.getScreenFragment().getActivity() != null) {
+        config.getScreenFragment().getActivity().setRequestedOrientation(config.getScreenOrientation());
+      }
+    }
+  }
+
+  private @Nullable ScreenStackHeaderConfig findHeaderConfig() {
+    ViewParent parent = getScreen().getContainer();
+    while (parent != null) {
+      if (parent instanceof Screen) {
+        ScreenStackHeaderConfig headerConfig = ((Screen) parent).getHeaderConfig();
+        if (headerConfig != null) {
+          return headerConfig;
+        }
+        parent = ((Screen) parent).getContainer();
+      } else {
+        parent = parent.getParent();
+      }
+    }
+    return null;
+  }
+
+  protected boolean hasChildScreenWithConfig(Screen screen) {
+    if (screen == null) {
+      return false;
+    }
+    for (ScreenContainer sc : screen.getFragment().getChildScreenContainers()) {
+      // we check only the top screen for header config
+      Screen topScreen = sc.getTopScreen();
+      ScreenStackHeaderConfig headerConfig = topScreen != null ? topScreen.getHeaderConfig(): null;
+      if (headerConfig != null) {
+        return true;
+      }
+      if (topScreen.getFragment().mHasChildWithConfig) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  protected void markParentFragments(boolean hasChildWithConfig) {
+    ViewParent parent = getScreen().getContainer();
+    while (parent != null) {
+      if (parent instanceof Screen) {
+        ((Screen) parent).getFragment().mHasChildWithConfig = hasChildWithConfig;
```

Comment:
If this does not result in `mHasChildWithConfig` change there is no point in updating other parents up the view hierarchy (there will be no updates as well) – we can break the loop if this occurs